src/app/layers/symbology/symbology-raster/symbology-raster-mapping-colorizer.component.ts

```diff
@@ -1,62 +1,94 @@
-import {Component, Input, Output, EventEmitter, ChangeDetectionStrategy, OnChanges, SimpleChanges} from '@angular/core';
-
-import {MappingColorizerRasterSymbology} from '../symbology.model';
+import {
+    Component,
+    Input,
+    Output,
+    EventEmitter,
+    ChangeDetectionStrategy,
+    OnChanges,
+    SimpleChanges,
+    OnInit,
+    AfterViewInit, OnDestroy, ElementRef
+} from '@angular/core';
+
+import {AbstractVectorSymbology, MappingColorizerRasterSymbology} from '../symbology.model';
 import {MatSliderChange} from '@angular/material';
 import {ColorizerData} from '../../../colors/colorizer-data.model';
 import {ColorBreakpoint} from '../../../colors/color-breakpoint.model';
+import {RasterLayer, VectorLayer} from '../../layer.model';
+import {BehaviorSubject, combineLatest as observableCombineLatest, Subscription} from 'rxjs';
+import {HistogramData} from '../../../plots/histogram/histogram.component';
+import {LayoutService} from '../../../layout.service';
+import {ProjectService} from '../../../project/project.service';
+import {Operator} from '../../../operators/operator.model';
+import {HistogramType} from '../../../operators/types/histogram-type.model';
+import {DataType} from '../../../operators/datatype.model';
+import {Unit} from '../../../operators/unit.model';
+import {first} from 'rxjs/operators';
+import {ResultTypes} from '../../../operators/result-type.model';
+import {MappingQueryService} from '../../../queries/mapping-query.service';
+import {MapService} from '../../../map/map.service';
 
 @Component({
     selector: 'wave-symbology-raster-mapping-colorizer',
     templateUrl: 'symbology-raster-mapping-colorizer.component.html',
     styleUrls: ['symbology-raster-mapping-colorizer.component.scss'],
     changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class SymbologyRasterMappingColorizerComponent implements OnChanges {
-
-    _symbology: MappingColorizerRasterSymbology;
+export class SymbologyRasterMappingColorizerComponent implements OnChanges, OnInit, AfterViewInit, OnDestroy {
+
+    @Input() layer: RasterLayer<MappingColorizerRasterSymbology>;
+    @Output() symbologyChanged: EventEmitter<MappingColorizerRasterSymbology> = new EventEmitter<MappingColorizerRasterSymbology>();
+
+    symbology: MappingColorizerRasterSymbology;
+
+    layerMinValue: number | undefined = undefined;
+    layerMaxValue: number | undefined = undefined;
+    layerHistogramData$: BehaviorSubject<HistogramData> = new BehaviorSubject(undefined);
```

Comment:
In dem Fall ReplaySubect(1)? Denn wenn noch nichts da ist, brauchst du das dennoch?

Reply:
done